### PR TITLE
Added support for github actions

### PR DESCRIPTION
**Pull request by MeghnaPrabhu**
*Description of changes:*
Enabling github actions requires these changes. 
This includes a golint.yml config that will be used in Makefile.
The other changes are lint errors that needs to be fixed for a successful build.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: fluent-bit-firehose.go [hunk 1]
```diff
@@ -50,9 +50,9 @@ func getPluginInstance(ctx unsafe.Pointer) *firehose.OutputPlugin {
 }
 
 // The "export" comments have syntactic meaning
-// This is how the compiler knows a function should be callable from the C code
+// This is how the compiler knows a function should be callable from the C code.
 
-//export FLBPluginRegister
+// FLBPluginRegister exporter.
```
Comment (hossain-rayhan) [inline fluent-bit-firehose.go:55]: I believe, the `export` word  in comment has a syntactic meaning. It's a cgo thing. I am not sure if it has been changed now.

**Review by sonofachamp — CHANGES_REQUESTED** [review]
I don't think GitHub actions setup should require any changes to `Makefile`. Where does the `golangci-lint` binary come from? Also as Rayhan pointed out, the `export` comments have significant meaning.

File: fluent-bit-firehose.go (hunk 1 above)
Comment (PettitWesley) [reply fluent-bit-firehose.go:55]: > I believe, the export word in comment has a syntactic meaning. It's a cgo thing.

💯

Comment (PettitWesley) [reply fluent-bit-firehose.go:55]: (And this is why we can't run Golint on this file)

Comment (MeghnaPrabhu) [reply fluent-bit-firehose.go:55]: Thanks for pointing this out. I wasn't aware that these comments had syntactic meaning. I will skip this file for golint checks.

**Comment by MeghnaPrabhu** [comment]
> I don't think GitHub actions setup should require any changes to `Makefile`. Where does the `golangci-lint` binary come from? Also as Rayhan pointed out, the `export` comments have significant meaning.

golangci-lint is a better and faster linter (https://golangci-lint.run/) which supports 48 other linters, golint is one of them. It could be useful if we decide to add more linters in the future. I do not see any major risks including this linter as I see it also being used in other aws repos. But I'm also okay with using the basic golint as well.

**Comment by sonofachamp** [comment]
> > I don't think GitHub actions setup should require any changes to `Makefile`. Where does the `golangci-lint` binary come from? Also as Rayhan pointed out, the `export` comments have significant meaning.
> 
> golangci-lint is a better and faster linter (https://golangci-lint.run/) which supports 48 other linters, golint is one of them. It could be useful if we decide to add more linters in the future. I do not see any major risks including this linter as I see it also being used in other aws repos. But I'm also okay with using the basic golint as well.

For now we can update the `Makefile` to install the existing `golint` dependency and add the GitHub actions workflow yaml file to get CI set up and look at improving our linter after.

**Comment by MeghnaPrabhu** [comment]
Opened a new PR. Closing this one.

File: Makefile [hunk 2]
```diff
@@ -23,7 +23,7 @@ PLUGIN_BINARY := ./bin/firehose.so
 build: $(PLUGIN_BINARY)
 
 $(PLUGIN_BINARY): $(SOURCES)
-	PATH=${PATH} golint ./plugins ./firehose
+	PATH=${PATH} ./bin/golangci-lint run -c golint.yml
```
Comment (PettitWesley) [inline Makefile:26]: My question/concern is- can you run and install this locally? The make targets are all meant to be runnable on a dev machine. The executable path makes me suspect it only works in GitHub actions..

**Pull request closed without merging**